frontend/src/components/modules/GameInterface/GameInterface.tsx

```diff
@@ -0,0 +1,258 @@
+
+  useEffect(() => {
+    setTimer(matchInfo.timerTime);
+  }, [matchInfo.timerTime]);
+
+  useEffect(() => {
+    let intervalId: NodeJS.Timeout | null = null;
+  
+    if (isTimerRunning) {
+      intervalId = setInterval(() => {
+        setTimer((prevTimer) => (prevTimer > 0 ? prevTimer - 1 : 0));
+      }, 1000);
+    } else {
+      if (intervalId !== null) {
+        clearInterval(intervalId);
+      }
+    }
+  
+    return () => {
+      if (intervalId !== null) {
+        clearInterval(intervalId);
+      }
+    };
+  }, [isTimerRunning, matchInfo.timerTime]);
+  
+
+  const buttonToTypeMap: Record<string, PointType> = {
+    M: "men",
+    K: "kote",
+    D: "do",
+    T: "tsuki",
+    "\u0394": "hansoku"
+  };
+
+  const selectedPointType = buttonToTypeMap[selectedButton];
+
+  const pointRequest: AddPointRequest = {
+    pointType: selectedPointType,
+    pointColor: playerColor
+  };
+
+  const handlePointShowing = async (): Promise<void> => {
+    setOpen(false);
+    if (matchId !== undefined) {
+      await apiPointRequest(matchId, pointRequest);
+    }
+  };
+
+  const handleRadioButtonClick = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ): void => {
+    setSelectedButton(event.target.value);
+  };
+
+  const handleOpen = (player: number): void => {
+    setSelectedButton("");
+    setOpen(true);
+    if (player === 1) {
+      setPlayerColor("white");
+    }
+    if (player === 2) {
+      setPlayerColor("red");
+    }
+  };
+
+  const apiPointRequest = async (
+    matchId: string,
+    body: AddPointRequest
+  ): Promise<void> => {
+    try {
+      await api.match.addPoint(matchId, body);
+    } catch (error) {
+      console.error("Data couldn't be sent", error);
+    }
+  };
+
+  const apiTimerRequest = async (matchId: string): Promise<void> => {
+    try {
+      if (!isTimerRunning) {
+        await api.match.startTimer(matchId);
+        setTimer(matchInfo.timerTime);
+      } else {
+        await api.match.stopTimer(matchId);
+      }
+    } catch (error) {
+      console.error("Data couldn't be sent", error);
+    }
+  };
+
+  const handleTimerChange = async (): Promise<void> => {
+    setIsTimerRunning((prevIsTimerRunning) => !prevIsTimerRunning);
+    if (matchId !== undefined) {
+      await apiTimerRequest(matchId);
+    }
+  };
+
+  function handleClose(): void {
+    setOpen(false);
+  }
+
+  return (
+    <div className="app-container">
+      <main className="main-content">
+        <Box display="flex" gap="20px" justifyContent="center">
+          <Box className="playerBox" bgcolor="white">
+            <Typography variant="h3">{matchInfo.playerNames[0]}</Typography>
+          </Box>
+          <Box className="playerBox" bgcolor="#db4744">
+            <Typography variant="h3">{matchInfo.playerNames[1]}</Typography>
+          </Box>
+        </Box>
+        <Box display="flex" gap="20px" justifyContent="center">
+          <Timer timer={timer} />
+          {userId !== null &&
```

Comment:
The underlying value for the userId is either undefined or string. No need for null checks (unless this is here to satisfy eslint or something)